src/tracing/traced_value.cc

```diff
@@ -0,0 +1,191 @@
+// Copyright 2016 the V8 project authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "tracing/traced_value.h"
+
+#include <math.h>
+#include <sstream>
+#include <stdio.h>
+#include <string>
+
+#if defined(_STLP_VENDOR_CSTD)
+// STLPort doesn't import fpclassify into the std namespace.
+#define FPCLASSIFY_NAMESPACE
+#else
+#define FPCLASSIFY_NAMESPACE std
+#endif
+
+namespace node {
+namespace tracing {
+
+namespace {
+
+void EscapeAndAppendString(const char* value, std::string* result) {
+  *result += '"';
+  char number_buffer[10];
+  while (*value) {
+    char c = *value++;
+    switch (c) {
+      case '\t':
+        *result += "\\t";
+        break;
+      case '\n':
+        *result += "\\n";
+        break;
+      case '\"':
+        *result += "\\\"";
+        break;
+      case '\\':
+        *result += "\\\\";
+        break;
+      default:
+        if (c < '\x20') {
+          snprintf(
+              number_buffer, arraysize(number_buffer), "\\u%04X",
+              static_cast<unsigned>(static_cast<unsigned char>(c)));
+          *result += number_buffer;
+        } else {
+          *result += c;
+        }
+    }
+  }
+  *result += '"';
+}
+
+std::string DoubleToCString(double v) {
+  switch (FPCLASSIFY_NAMESPACE::fpclassify(v)) {
+    case FP_NAN: return "\"NaN\"";
+    case FP_INFINITE: return (v < 0.0 ? "\"-Infinity\"" : "\"Infinity\"");
+    case FP_ZERO: return "0";
+    default:
+      // This is a far less sophisticated version than the one used inside v8.
+      std::ostringstream stream;
+      stream.imbue(std::locale("C"));  // Ignore locale
+      stream << v;
+      return stream.str();
```

Comment:
Does `return std::to_string(v);` do the trick here too, for the `default` case?

Reply:
I'll double check

Reply:
Ah, right, the challenge with `std::to_string(v)` is that it ignores any reasonable precision on the double and always prints six decimals, whereas the version used here will print with actual scientific notation (e.g. `std::to_string(v)` when `v = 1.23e7`, will print `12300000.000000` whereas the version used here will print `1.23e+07` as one would more reasonably expect.